File: baseLib/src/test/java/me/ycdev/android/lib/common/type/BooleanHolderTest.java
package me.ycdev.android.lib.common.type;

import org.junit.Test;

import static org.hamcrest.core.Is.is;
import static org.junit.Assert.assertThat;

public class BooleanHolderTest {
    @Test
    public void basic() {
        {
            BooleanHolder holder = new BooleanHolder(true);
            assertThat(holder.value, is(true));
        }

        {
            BooleanHolder holder = new BooleanHolder(false);
            assertThat(holder.value, is(false));
        }
    }
}

File: baseLib/src/test/java/me/ycdev/android/lib/common/type/BooleanHolderTest.kt
package me.ycdev.android.lib.common.type

import com.google.common.truth.Truth.assertThat
import org.junit.Test

class BooleanHolderTest {
    @Test
    fun basic() {
        run {
            val holder = BooleanHolder(true)
            assertThat(holder.value).isTrue()
        }

        run {
            val holder = BooleanHolder(false)
            assertThat(holder.value).isFalse()
        }
    }
}
